### ISSUE TITLE
Support for xmodmap modifiers?

### ISSUE THREAD

**Issue by yaxunema95** [autokey/autokey#76]
## Classification:

Enhancement

## Notes:

Currently, the modifiers available in AutoKey are Control, Alt, Shift, Super, Hyper, and Meta.

Xkb/xmodmap allows one to map keys to shift, lock, control, mod1, mod2, mod3, mod4, and mod5.

I use, for example, CAPS remapped to Scroll_Lock as mod3 in an Xkb layout to give me Vi-like keys when I hold down CAPS.

The problem is, if I want to map mod3+u to ctrl+z for undo, I have to remap CAPS from Scroll_Lock to either Meta or Hyper for use in hotkeys in AutoKey, which does not understand Scroll_Lock as a modifier. This causes issues with some applications. Urxvt interprets Meta, and Firefox and Thunderbird interpret both Meta and Hyper outside of editable text boxes (Scroll_Lock+BackSpace is problematic in Mozilla apps, too--it looks like their devs put some weird keyboard stuff in there that overrides Xorg).

I know mine is an odd use-case, but most window managers understand the mod1-5 notation. Could AutoKey? Another alternative would be for AutoKey to interpret Caps, Scroll, and Num Locks, and maybe Level3/5_Shift and Mode_switch as modifiers.

**Comment by xircon** [comment]
Can I recommend xcape? e.g. in i3wm to map tap super key to change workspace:
```
exec_always xcape -e 'Super_L=Alt_L|Tab'
```
so outside i3 it would be just xcape -e 'Super_L=Alt_L|Tab'

This remaps left super to alt+tab.

https://github.com/alols/xcape has a brief write up that does not do it justice.

**Comment by yaxunema95** [comment]
Xcape is a good program, but it's purpose is to treat modifiers as normal keys when pressed and released by themselves.

What I need to be able to do is use *additional modifiers* for hotkeys in AutoKey beyond Control, Shift, Super, Meta, and Hyper, perhaps in the format Mod1, Mod2, Mod3, Mod4, Mod5.

I found a kludgy workaround by using xbindkeys to grab the Mod3 keybindings to use autokey-run to perform the phrases (Openbox can grab the keybindings but won't run the commands if I use Mod3, probably because it isn't mapped as Control, Shift, Super, Meta, or Hyper--I stand corrected on my statement about window managers). Of course, this isn't optimal since I now have two programs running that handle keybindings (in addition to my window manager).

Another solution might be to just use Hyper instead of Scroll_Lock, but attempt to make window filters in AutoKey for apps like Firefox and Thunderbird that don't comply. My fear is it could add delay to the simpler keys in the layout (stuff in Xkb that doesn't require AutoKey, such as Hyper+h,j,k,l for left, down, up, and right) in all other apps since AutoKey may grab the keys first, then check if the window is in the filter.

In other words, if I could choose Mod3 for hotkeys in AutoKey, I could keep the lower-level Xkb stuff while using AutoKey to pass keyboard shortcuts to applications without having to use workarounds.

**Comment by josephj11** [comment]
@xircon and @yaxunema95 I hadn't noticed this issue before. Is there still any interest in it or can it be closed as won't fix?

Since this was filed, we did add an option for AutoKey to ignore the CapsLock key so it could be remapped and work as expected, but we have not done anything further - and probably won't unless someone volunteers to work on it.

You may also want to take a look at this [article](https://github.com/autokey/autokey/wiki/Advanced-Topics) in our wiki which might help with this issue.

**Comment by xircon** [comment]
I no longer use autokey, feel free.

**Comment by josephj11** [comment]
@xircon Sorry to hear that. Thanks for replying.
